tests/integration/release/test_release.py

```diff
@@ -14,131 +15,114 @@
 from pseudopeople.configuration import Keys, get_configuration
 from pseudopeople.configuration.entities import NO_NOISE
 from pseudopeople.configuration.noise_configuration import NoiseConfiguration
+from pseudopeople.constants import paths
 from pseudopeople.constants.metadata import DatasetNames
 from pseudopeople.constants.noise_type_metadata import (
     GUARDIAN_DUPLICATION_ADDRESS_COLUMNS,
 )
 from pseudopeople.dataset import Dataset
 from pseudopeople.entity_types import ColumnNoiseType, RowNoiseType
-from pseudopeople.interface import generate_decennial_census, generate_social_security
+from pseudopeople.filter import get_generate_data_filters
+from pseudopeople.interface import (
+    generate_social_security,
+    get_dataset_filepaths,
+    validate_source_compatibility,
+)
+from pseudopeople.loader import load_standard_dataset
 from pseudopeople.noise_entities import NOISE_TYPES
 from pseudopeople.noise_functions import merge_dependents_and_guardians
 from pseudopeople.schema_entities import COLUMNS, DATASET_SCHEMAS
+from pseudopeople.utilities import DASK_ENGINE, get_engine_from_string
 from tests.integration.conftest import SEED, _get_common_datasets
+from tests.integration.release.conftest import (
+    DATASET_ARG_TO_FULL_NAME_MAPPER,
+    RI_FILEPATH,
+)
+from tests.integration.release.utilities import (
+    run_do_not_respond_tests,
+    run_guardian_duplication_tests,
+    run_omit_row_tests,
+)
 from tests.utilities import (
     get_single_noise_type_config,
     initialize_dataset_with_sample,
     run_column_noising_tests,
 )
 
+ROW_TEST_FUNCTIONS = {
+    "omit_row": run_omit_row_tests,
+    "do_not_respond": run_do_not_respond_tests,
+    "duplicate_with_guardian": run_guardian_duplication_tests,
+}
 
-def test_column_noising(
-    unnoised_dataset: Dataset,
-    noised_data: pd.DataFrame,
-    config: dict[str, Any],
-    dataset_name: str,
+
+def test_release_row_noising(
+    dataset_params: tuple[
+        str,
+        Callable[..., pd.DataFrame],
+        Path | str | None,
+        int | None,
+        str | None,
+        Literal["pandas", "dask"],
+    ],
     fuzzy_checker: FuzzyChecker,
 ) -> None:
-    """Tests that columns are noised as expected"""
-    check_noised, check_original, shared_idx = _get_common_datasets(
-        unnoised_dataset, noised_data
-    )
+    dataset_name, _, source, year, state, engine_name = dataset_params
+    full_dataset_name = DATASET_ARG_TO_FULL_NAME_MAPPER[dataset_name]
+    dataset_schema = DATASET_SCHEMAS.get_dataset_schema(full_dataset_name)
+    config = get_configuration()
 
-    run_column_noising_tests(
-        dataset_name, config, fuzzy_checker, check_noised, check_original, shared_idx
-    )
+    # update parameters
+    if source is None:
+        source = paths.SAMPLE_DATA_ROOT
+    elif isinstance(source, str) or isinstance(source, Path):
+        source = Path(source)
+        validate_source_compatibility(source, dataset_schema)
```

Comment:
nit: Can you add a docstring to `validate_source_compatibility`? I know it's not a new thing in this PR but it would have been helpful when reviewing.